uvtools/dspec.py

```diff
@@ -44,67 +53,157 @@ def calc_width(filter_size, real_delta, nsamples):
-                             skip_wgt=0.1, maxiter=100, gain=0.1, **win_kwargs):
-    '''Apply a highpass fourier filter to data. Uses aipy.deconv.clean. 
+def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False, tol=1e-9, window='none', 
+                             skip_wgt=0.1, maxiter=100, gain=0.1, alpha=0.2, filt2d_mode='rect',
+                             edgecut=0):
+    '''Apply a highpass fourier filter to data. Uses aipy.deconv.clean. Default is a 1D clean
+    on the last axis of data.
 
     Arguments:
-        data: 1D or 2D (real or complex) numpy array to be filtered along the last dimension.
+        data: 1D or 2D (real or complex) numpy array to be filtered.
             (Unlike previous versions, it is NOT assumed that weights have already been multiplied
             into the data.)
         wgts: real numpy array of linear multiplicative weights with the same shape as the data. 
         filter_size: the half-width (i.e. the width of the positive part) of the region in fourier 
             space, symmetric about 0, that is filtered out. In units of 1/[real_delta].
-        real_delta: the bin width in real space of the dimension to be filtered
+            Alternatively, can be fed as len-2 tuple specifying the negative and positive bound
+            of the filter in fourier space respectively (for asymmetric CLEANing).
+            For 2D cleaning, a len-2 list must be fed with each element following the rules
+            above for each CLEAN dimension.
+        real_delta: the bin width in real space of the dimension to be filtered.
+            If 2D cleaning, then real_delta must also be a len-2 list.
+        clean2d : bool, if True perform 2D clean, else perform a 1D clean on last axis.
         tol: CLEAN algorithm convergence tolerance (see aipy.deconv.clean)
         window: window function for filtering applied to the filtered axis. 
-            See aipy.dsp.gen_window for options.
+            See aipy.dsp.gen_window for options. If clean2D, can be fed as a list
+            specifying the window for each axis in data.
         skip_wgt: skips filtering rows with very low total weight (unflagged fraction ~< skip_wgt).
             Model is left as 0s, residual is left as data, and info is {'skipped': True} for that 
             time. Only works properly when all weights are all between 0 and 1.
         maxiter: Maximum number of iterations for aipy.deconv.clean to converge.
         gain: The fraction of a residual used in each iteration. If this is too low, clean takes
             unnecessarily long. If it is too high, clean does a poor job of deconvolving.
-        win_kwargs : any keyword arguments for the window function selection in aipy.dsp.gen_window.
-            Currently, the only window that takes a kwarg is the tukey window with a alpha=0.5 default.
+        alpha : float, if window is 'tukey', this is its alpha parameter.
```

Comment:
I think I'd prefer to hold off on this change. It breaks thing downstream and I think win_kwargs is more extensible in the future if we want to try other window functions besides tukey that take parameters.

Reply:
Okay, well one thing I've been meaning to do is to transition from using `aipy.dsp.gen_window` which only holds a handful of window functions, and to transition to use `scipy.signal.windows`, which holds a lot more window functions we could explore using. I added a `gen_window` function in `dspec` to do this, and I think that this change + getting rid of `win_kwargs` doesn't actually break things downstream.. but I'll investigate more and get back to you on that.